crates/core_arch/src/aarch64/neon/mod.rs

```diff
@@ -795,6 +796,292 @@ pub unsafe fn vld1q_f64(ptr: *const f64) -> float64x2_t {
     transmute(f64x2::new(*ptr, *ptr.offset(1)))
 }
 
+/// Store multiple single-element structures from one, two, three, or four registers.
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(str))]
+#[allow(clippy::cast_ptr_alignment)]
+pub unsafe fn vst1_s8(ptr: *mut i8, a: int8x8_t) {
+    copy_nonoverlapping(
+        &a as *const int8x8_t as *const u8,
+        ptr as *mut u8,
+        size_of::<int8x8_t>(),
+    )
+}
+
+/// Store multiple single-element structures from one, two, three, or four registers.
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(str))]
+#[allow(clippy::cast_ptr_alignment)]
+pub unsafe fn vst1q_s8(ptr: *mut i8, a: int8x16_t) {
+    copy_nonoverlapping(
+        &a as *const int8x16_t as *const u8,
+        ptr as *mut u8,
+        size_of::<int8x16_t>(),
+    )
+}
+
+/// Store multiple single-element structures from one, two, three, or four registers.
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(str))]
+#[allow(clippy::cast_ptr_alignment)]
+pub unsafe fn vst1_s16(ptr: *mut i16, a: int16x4_t) {
+    copy_nonoverlapping(
+        &a as *const int16x4_t as *const u8,
+        ptr as *mut u8,
+        size_of::<int16x4_t>(),
+    )
+}
+
+/// Store multiple single-element structures from one, two, three, or four registers.
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(str))]
+#[allow(clippy::cast_ptr_alignment)]
+pub unsafe fn vst1q_s16(ptr: *mut i16, a: int16x8_t) {
+    copy_nonoverlapping(
+        &a as *const int16x8_t as *const u8,
```

Comment:
I checked with Clang and these aren't actually fully unaligned stores. Instead the pointers are required to be aligned to the element size (`i16`, `i32`, `i64`, etc). You can do the same here by using that type as the element for `copy_nonoverlapping` instead of `u8`.

Reply:
Fixed as adviced.